deployability/modules/testing/testing.py

```diff
@@ -22,21 +23,49 @@ def run(cls, payload: InputPayload) -> None:
         """
         payload = InputPayload(**dict(payload))
-        inventory = Inventory(**Utils.load_from_yaml(payload.inventory))
-        logger.info(f"Running tests for {inventory.ansible_host}")
         extra_vars = cls._get_extra_vars(payload).model_dump()
+        targets_paths = payload.targets
+        for target_path in targets_paths:
+            path = ', '.join(list(eval(target_path).values()))
+            target = Inventory(**Utils.load_from_yaml((path)))
+            logger.info(f"Running tests for {target.ansible_host}")  
+        targets = {}
+        for item in targets_paths:
+            dictionary = eval(item)
+            targets.update(dictionary)
+        target_string = json.dumps(targets)
+        extra_vars['targets'] = target_string.replace('"',"")
+        dependencies_paths = payload.dependencies
+        for dependency_path in dependencies_paths:
+            path = ', '.join(list(eval(dependency_path).values()))
+            dependency = Inventory(**Utils.load_from_yaml((path)))
+            logger.info(f"Dependencies {dependency.ansible_host}")  
+        dependencies = {}
+        for item in targets_paths:
+            dictionary = eval(item)
+            dependencies.update(dictionary)
+        target_string = json.dumps(dependencies)
+        extra_vars['dependencies'] = target_string.replace('"',"")
+        extra_vars['local_host_path'] = str(Path(__file__).parent.parent.parent)
+        extra_vars['current_user'] = os.getlogin()
         logger.debug(f"Using extra vars: {extra_vars}")
-        dependencies_dict = {}
-        for dependency in extra_vars['dependencies']:
-            dependency = ast.literal_eval(dependency)
-            dependencies_dict.update(dependency)
-        extra_vars['dependencies'] = dependencies_dict
-        ansible = Ansible(ansible_data=inventory.model_dump())
-        cls._setup(ansible, extra_vars['working_dir'])
+        for target in targets_paths:
+            target_value = eval(target).values()
+            target_inventory = Inventory(**Utils.load_from_yaml(str(list(target_value)[0])))
+            ansible = Ansible(ansible_data=target_inventory.model_dump())
+            cls._setup(ansible, extra_vars['working_dir'])
+
+        target_inventory = Inventory(**Utils.load_from_yaml(str(list(eval(targets_paths[0]).values())[0])))
+        ansible = Ansible(ansible_data=target_inventory.model_dump())
         cls._run_tests(payload.tests, ansible, extra_vars)
-        if payload.cleanup:
-            logger.info("Cleaning up")
-            cls._cleanup(ansible, extra_vars['working_dir'])
+
+        for target in targets_paths:
+            target_value = eval(target).values()
+            target_inventory = Inventory(**Utils.load_from_yaml(str(list(target_value)[0])))
+            if payload.cleanup:
+                logger.info("Cleaning up")
+                cls._cleanup(ansible, extra_vars['working_dir'])
+
```

Comment:
It seems ok, but couldn't this be separated in a few functions? or simplify the code in some way?

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/5112/commits/dd818b765d551ab2c6431656986ec60cea05225c